scripts/check_valgrind.sh

```diff
@@ -23,7 +23,9 @@
 #
 # This script runs Valgrind on an input binary paths with specified memory leak detection flags. 
 # The Valgrind output is parsed. If any errors are found, they are printed and the script exits with a status of 1.
-# As a second argument, you can provide a path to a suppression file that is used by Valgrind to suppress certain errors.
+# If errors are found, the Valgrind output is kept in the file valgrind-output.log for further inspection.
+# Using "--supp=", you can provide a path to a suppression file that is used by Valgrind to suppress certain errors.
+# With "--ntasks=", you can provide the number of processes to use with mpi (default is 1).
 #
 
```

Comment:
I suggest to add a help string here that is printed whenever an input error occurs.

Something like

USAGE="This script executes valgrind in parallel on a given input file.\n\n
$0 [FILE] --supp=[SUPPRESSION_FILE] --ntasks=[NUM_TASKS]\n\n

while run valgrind on FILE. Optionally you can provide a suppression file and the number of parallel processes to use with MPI.
"